website/docs/tutorials/react-rsc.md

```diff
@@ -0,0 +1,198 @@
+
+1. given `lang`, take an I18n instance and store it in the [`cache`](https://react.dev/reference/react/cache) so it can be used server-side
+2. given `lang`, take an I18n instance and make it available to client components via I18nProvider
+
+This is how step (1) can be implemented:
+
+```tsx title="src/app/[lang]/layout.tsx"
+import { setI18n } from "@lingui/react/server";
+import { allI18nInstances } from "./appRouterI18n";
+import { LinguiClientProvider } from "./LinguiClientProvider";
+
+type Props = {
+  params: {
+    lang: string;
+  };
+  children: React.ReactNode;
+};
+
+export default function RootLayout({ params: { lang }, children }: Props) {
+  const i18n = allI18nInstances[lang]; // get a ready-made i18n instance for the given locale
+  setI18n(i18n); // make it available server-side for the current request
+
+  return (
+    <html lang={lang}>
+      <body>
+        <LinguiClientProvider initialLocale={lang} initialMessages={i18n.messages}>
+          <YourApp />
+        </LinguiClientProvider>
+      </body>
+    </html>
+  );
+}
+```
+
+Step (2) is implemented in `LinguiClientProvider`, which is a client component:
+
+```tsx title="LinguiClientProvider.tsx"
+"use client";
+
+import { I18nProvider } from "@lingui/react";
+import { type Messages, setupI18n } from "@lingui/core";
+import { useState } from "react";
+
+export function LinguiClientProvider({
+  children,
+  initialLocale,
+  initialMessages,
+}: {
+  children: React.ReactNode;
+  initialLocale: string;
+  initialMessages: Messages;
+}) {
+  const [i18n] = useState(() => {
+    return setupI18n({
+      locale: initialLocale,
+      messages: { [initialLocale]: initialMessages },
+    });
+  });
+  return <I18nProvider i18n={i18n}>{children}</I18nProvider>;
+}
+```
+
+:::tip
+Why we are not passing the I18n instance directly from `RootLayout` to the client via `LinguiClientProvider`? It's because the I18n object isn't serializable, and cannot be passed from server to client.
+:::
+
+Lastly, there's the `appRouterI18n.ts` file, which is only executed on server and holds one instance of I18n object for each locale of our application. See [here](#) how it's implemented in the example.
+
+### Rendering translations in server and client components
+
+Below you can see an example of a React component. This component can be rendered **both with RSC and on client**. This is great if you're migrating a Lingui-based project from pages router to app router because you can keep the same components working in both worlds.
+
+In fact, if you swapped the html tags for their more universal alternatives, this component could also be used in React Native.
+
+```tsx title="app/[lang]/components/SomeComponent.tsx"
+import { Trans, t } from "@lingui/macro";
+
+export function SomeComponent() {
+  const { i18n } = useLingui();
+  return (
+    <div>
+      <p>
+        <Trans>Some Item</Trans>
+      </p>
+      <p>{t(i18n)`Other Item`}</p>
+    </div>
+  );
+}
```

Comment:
may be it's worth to  add an import statement for useLingui / Trans to avoid confusion